engine-precompiles/src/random.rs

```diff
@@ -0,0 +1,85 @@
+use super::{EvmPrecompileResult, Precompile};
+#[cfg(feature = "contract")]
+use crate::prelude::sdk;
+use crate::prelude::Address;
+#[cfg(not(feature = "contract"))]
+use crate::prelude::H256;
+use crate::PrecompileOutput;
+use evm::{Context, ExitError};
+
+mod costs {
+    use crate::prelude::types::Gas;
+
+    // TODO(#51): Determine the correct amount of gas
+    pub(super) const RANDOM_BYTES_GAS: Gas = 0;
+}
+
+const ERR_TARGET_TOKEN_NOT_FOUND: &str = "Target token not found";
+
+pub struct RandomBytes; //TransferEthToNear
+
+impl RandomBytes {
+    /// Random bytes precompile address
+    ///
+    /// Address: `0xf861511815955326b953fa97b6955a2f8020a4e9`
+    /// This address is computed as: `&keccak("randomBytes")[12..]`
+    pub const ADDRESS: Address =
+        super::make_address(0xf8615118, 0x15955326b953fa97b6955a2f8020a4e9);
+}
+
+impl Precompile for RandomBytes {
+    fn required_gas(_input: &[u8]) -> Result<u64, ExitError> {
+        Ok(costs::RANDOM_BYTES_GAS)
+    }
+
+    #[cfg(feature = "contract")]
```

Comment:
Now that we have the standalone engine I think it is good to try to avoid implementations which require NEAR host functions. In this case we can simply store the seed as a field in the `RandomBytes` struct. This pushes the NEAR call up one level to the precompile initialization. Similarly there we can have the random seed be an input to creating precompiles, pushing it back another level. Ultimately the host function is only called at the boundary of the application (`engine/lib.rs` in the wasm contract), or in the case of the standalone engine we could derive the seed from the NEAR block information, and the value is passed down into the core logic.

I think it would be good if we could eliminate this `contract` feature from the precompiles crate entirely. This requires a little refactoring of the native precompiles implementation, but I think it is doable.

Reply:
Done.